kernel-modules/probe/collector_probe.c

```diff
@@ -259,15 +410,172 @@ static __always_inline int enter_probe(long id, struct sys_enter_args* ctx) {
 
   sc_evt = get_syscall_info(id);
   if (sc_evt == NULL || (sc_evt->flags & UF_USED) == 0) {
+#ifdef CAPTURE_SOCKETCALL
+    if (id != __NR_socketcall)
+      return 0;
+    // call_filler() and handle_socketcall() will change evt_type/drop_flags
+    // based on the detected socket call
+    evt_type = PPME_GENERIC_E;
+    drop_flags = UF_ALWAYS_DROP;
+#else
     return 0;
+#endif
+  } else {
+    evt_type = sc_evt->enter_event_type;
+    drop_flags = sc_evt->flags;
   }
 
-  evt_type = sc_evt->enter_event_type;
-  drop_flags = sc_evt->flags;
-
-  // To satisfy some verifier requiremnts in later parts of the falco plumbing/fillers,
-  // it is necessary to copy the context onto the stack.
-  memcpy(stack_ctx.args, _READ(ctx->args), sizeof(unsigned long) * NUM_SYS_ENTER_ARGS);
+  /* Syscall tracepoints follow their own format (=structure) of arguments.
+   * Copying all arguments, e.g. with
+   *   memcpy(stack_ctx.args, _READ(ctx->args), sizeof(unsigned long) * NUM_SYS_ENTER_ARGS);
+   * would copy beyond the format and results in an EACCESS triggered by
+   * format/structure offset checking in the perf event subsystem when installing
+   * the BPF program.
+   * Also the system exit fillers require access to the arguments to fill event
+   * data.  Note that the syscall tracepoints do not provide any argument data
+   * in the syscall exit format.  Of course, trying to access argument data in
+   * filler results in BPF verifier errors.
+   *
+   * To provide arguments to the syscall enter and exit fillers, extract the
+   * arguments from syscall specific formats/structures to an sys_enter_args
+   * structure and use a BPF map to make them available to the enter/exit
+   * fillers.
+   */
+  switch (id) /* maybe use sc_evt here..... */
```

Comment:
Could we move this switch statement to its own inline function in a separate header, and deconstruct each of the cases into their own inline functions as well to keep it relatively clean. My reasoning is primarily to keep this function clear of the implementation details of the copying

Reply:
Hi @Stringy , many thanks for the review!

Yes, I can move the structure definitions to a separate header file. I would then also extract and add the inline function comprising of the switch statement. Also no real objections on splitting the cases into inline functions for doing the copying.

Reply:
handled in PR https://github.com/stackrox/collector/pull/1072